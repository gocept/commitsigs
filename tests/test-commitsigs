#!/bin/sh

"$TESTDIR/hghave" gpg openssl || exit 80

hgverifysigs () {
    hg verifysigs > output
    echo "% hg verifysigs (exit code: $?)"
    sed 's|:[0-9a-f]\+:|:XXXXXXXXXXXX:|' < output
    rm output
}

hg init
echo "Hello" > a.txt
hg add a.txt
echo "% Commit with no signature"
hg commit -m "Unsigned" -d '1000 0'

cat >> $HGRCPATH <<EOF
[extensions]
commitsigs = $TESTDIR/../commitsigs.py

[commitsigs]
gnupg.flags = --no-permission-warning --no-secmem-warning \
              --homedir $TESTDIR/gpg

EOF

hgverifysigs

echo "World" >> a.txt
echo "% Commit with GnuPG signature"
hg commit -m "GnuPG" -d '2000 0'
hgverifysigs

cat >> $HGRCPATH <<EOF
[commitsigs]
scheme = openssl
openssl.pubkey = $TESTDIR/openssl/alice.pub.pem
openssl.seckey = $TESTDIR/openssl/alice.sec.pem
EOF

echo "!" >> a.txt
echo "% Commit with OpenSSL signature"
hg commit -m "OpenSSL" -d '3000 0'

hgverifysigs

cat >> $HGRCPATH <<EOF
[commitsigs]
scheme = x
EOF

echo "% unknown scheme"
hg status

true
