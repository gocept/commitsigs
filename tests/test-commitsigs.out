% Commit with no signature
% hg verifysigs (exit code: 1)
0:XXXXXXXXXXXX: ** no signature
% Commit with GnuPG signature
% hg verifysigs (exit code: 1)
0:XXXXXXXXXXXX: ** no signature
1:XXXXXXXXXXXX: good gnupg signature
% Commit with OpenSSL signature
% hg verifysigs (exit code: 1)
0:XXXXXXXXXXXX: ** no signature
1:XXXXXXXXXXXX: good gnupg signature
2:XXXXXXXXXXXX: good openssl signature
% unknown scheme
abort: unknown signature scheme: x
